libs/facilities/backend/timeseries/src/lib/services/timeseries.service.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { faker } from '@faker-js/faker';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Prisma } from '@prisma/client';
+import { PrismaService } from 'common-backend-prisma';
+import { lastValueFrom } from 'rxjs';
+
+import { IGetTimeSeriesParams, IGetTimeseriesQuery } from '../interfaces/request.interface';
+import { XdTimeseriesService } from './timeseries.service';
+
+describe('TimeseriesService', () => {
+	let service: XdTimeseriesService;
+	let prisma: PrismaService = new PrismaService();
+
+	beforeEach(async () => {
+		const prismaServiceMock = {
+			onModuleInit: jest.fn(),
+
+			selectKeysFromJSON: prisma.selectKeysFromJSON,
+			isTTimeSeriesData: prisma.isTTimeSeriesData,
+			timeSeriesDataItem: {
+				findMany: jest.fn().mockImplementation(() => [
+					{
+						time: new Date(),
+						data: JSON.stringify({ test: 'test', test2: 'test2' }),
+					},
+				]),
+			},
+		};
+
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [
+				XdTimeseriesService,
+				{
+					provide: PrismaService,
+					useValue: prismaServiceMock,
+				},
+			],
+		}).compile();
+
+		service = module.get<XdTimeseriesService>(XdTimeseriesService);
+		prisma = module.get<PrismaService>(PrismaService);
+	});
+
+	afterAll(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+
+	it('should return the correct time series data', async () => {
+		const flow = faker.string.sample();
+		const presure = faker.string.sample();
+		const findManyResult = {
+			time: new Date(),
+			timeSeriesItementityId: faker.string.uuid(),
+			timeSeriesItempropertySetName: faker.string.sample(),
+			data: JSON.stringify({ flow: flow, presure: presure }) as Prisma.JsonValue,
+		};
+
+		const findManySpy = jest
+			.spyOn(prisma.timeSeriesDataItem, 'findMany')
+			.mockResolvedValue([findManyResult]);
+
+		const params: IGetTimeSeriesParams = {
+			entityId: findManyResult.timeSeriesItementityId,
+			propertySetName: findManyResult.timeSeriesItempropertySetName,
+		};
+
+		const result = await lastValueFrom(
+			service.getTimeSeriesFromDB({
+				...params,
+				select: ['flow', 'presure'],
+			}),
+		);
+
+		expect(findManySpy).toHaveBeenCalledTimes(1);
+
+		expect(result).toEqual([
+			{
+				time: expect.any(Date),
```

Comment:
Would be more consistent to check the exact date here

Reply:
true